### ISSUE TITLE
delete the ability to access a function parameter name from reflection

### ISSUE THREAD

**Issue by andrewrk** [ziglang/zig#6534]
Similar reasoning as ability to access doc comments with reflection. Too many ways to make confusing code. We don't want people reading code to have to wonder in the back of their minds if naming a function parameter a certain way would have a meaningful change.

related: #6530

**Comment by mlugg** [comment]
Isn't this already impossible? @andrewrk

**Comment by nektro** [comment]
yes since https://github.com/ziglang/zig/pull/10753

**Comment by amethys7dev** [comment]
@andrewrk I disagree with the decision here to remove function parameter names from being available via reflection.

I'm working on an Actor model system in Zig that takes an input struct type like the following:
```zig
const BankAccount = struct {
    balance: u64 = 0,

    pub fn deposit(self: *BankAccount, amount: u64) void {
        self.balance += amount;
    }

    pub fn withdraw(self: *BankAccount, amount: u64) void {
        self.balance -= amount;
    }
};
```

I need to create a mailbox for these functions that stores the function arguments so that I can delay their execution until desired.

This is what my desired DX would look like:
```zig
const BankAccountActor = Actor(BankAccount);
const new_actor = BankAccountActor.init();

// Any of these approaches work for me:
new_actor.sendMessage(BankAccountActor.Deposit { .amount = 100 });
new_actor.sendMessage("deposit", 100); // amount: u64
new_actor.sendDepositMessage(100); // amount: u64

new_actor.executeAllMessages();
```

I am not looking to change my behavior depending on the name of the function parameter, but simply pass it through. With no ability to get parameter names, I would have to label them generically as `param1`, `param2`, etc., making the code more difficult to use and reason about.

Is there a better way to accomplish what I'm hoping for? I would prefer to keep excess boilerplate in the BankAccount struct to a minimum, as it adds mental overhead to filter it out.
This is my "minimum viable example": [main.zig](https://github.com/user-attachments/files/21923929/main.txt)

**Comment by mlugg** [comment]
Your first usage example is precisely the kind of thing we want to avoid. Parameter names are not meant to be part of a function's "public API"; changing them should not affect whether code does or does not compile. However:

> ```zig
> new_actor.sendMessage("deposit", 100); // amount: u64
> new_actor.sendDepositMessage(100); // amount: u64
> ```

These two potential usages clearly don't depend on the parameter name at all? I'm not really sure why you think you need it.

If you want the callers to have to write the name, just make your function accept a `struct`, as in:
```zig
    pub fn deposit(self: *BankAccount, args: struct { amount: u64 }) void {
        self.balance += args.amount;
    }
```

This language decision is more-or-less set in stone. If you want further advice on how to design your API, please direct your questions to a Zig [community](https://github.com/ziglang/zig/wiki/Community).

**Comment by amethys7dev** [comment]
@mlugg Oh. In hindsight that makes infinitely more sense than what I was trying to do. Apologies for wasting your time, thank you!

As for the reasoning behind needing the name for the second two usages, it was mostly for IDE auto-suggestions and more clear error messages. But now I see that adding a type for the args makes way more sense
